src/containers/MainPlotContainer/index.tsx

```diff
@@ -116,6 +131,40 @@ class MainPlotContainer extends React.Component<MainPlotContainerProps> {
         });
     }
 
+    private updateThumbnails(cellId: string, url: string | null) {
+        this.setState((state) => {
+            // Move the updated thumbnail to the front of the map entries
+            const mapCopy = new Map(state.cellIdToZarrThumbnailUrl);
+            mapCopy.delete(cellId);
+            const mapEntries: [string, string | null][] = [
+                [cellId, url],
+                ...Array.from(mapCopy.entries()),
+            ];
+            // Limit the number of thumbnails stored in memory
+            const newEntries = mapEntries.slice(0, MAX_GENERATED_THUMBNAILS);
+            const newMap = new Map(newEntries);
+            return { cellIdToZarrThumbnailUrl: newMap };
+        });
+    }
+
+    private async loadThumbnailForZarr(cellId: string, srcPath: string | undefined): Promise<void> {
+        if (this.state.cellIdToZarrThumbnailUrl.has(cellId)) {
+            return;
+        } else if (this.state.cellIdToZarrThumbnailUrl.get(cellId) === null) {
+            // already loading, or load failed
+            return;
+        }
+        this.updateThumbnails(cellId, null);
+        if (srcPath && srcPath.endsWith(".ome.zarr")) {
+            const src = await createThumbnailImageSrc(srcPath);
+            this.updateThumbnails(cellId, src);
```

Comment:
Should there be a check here for success? Looks like `ome-zarr.js` does [relatively little validation](https://github.com/BioNGFF/ome-zarr.js/blob/main/src/render.ts) and largely [assumes the zarr contains at least some OME metadata](https://github.com/BioNGFF/ome-zarr.js/blob/764db3032d2a05204a40ad918edabdd070270798/src/utils.ts#L215). Might be worth a `try`/`catch`?